### DISCUSSION TITLE
linux - plugins.xml - run musescore error (appimage)

### DISCUSSION THREAD

**Discussion by aum7** [Audiveris/audiveris#764] [Q&A]
hi
thx for sharing, a fine piece of app

on linux (mint) i added 'plugins.xml' file into appropriate folder
``` xml
<plugins>
    <plugin id="MuseScore" tip="Invoke MuseScore on score XML">
       <!-- <arg>sh</arg> -->
        <!-- <arg>-c</arg> -->
        <arg>/home/myfolder/apps/musescore/MuseScore-Studio-4.4.4.243461245-x86_64.AppImage</arg>
        <arg>{}</arg>
    </plugin>
</plugins>
```

when i run 'default plugin' i get an error :

``` sh
Exporting sheet(s): [#1]

Score m2-036__ exported to /home/aumhren/.var/app/org.audiveris.audiveris/data/AudiverisLtd/audiveris/m2-036__/m2-036__.mxl

Process: [/home/aumhren/apps/musescore/MuseScore-Studio-4.4.4.243461245-x86_64.AppImage, /home/aumhren/.var/app/org.audiveris.audiveris/data/AudiverisLtd/audiveris/m2-036__/m2-036__.mxl]

Error launching java.io.IOException: Cannot run program "/home/aumhren/apps/musescore/MuseScore-Studio-4.4.4.243461245-x86_64.AppImage": error=2, No such file or directory {}Plugin{id:"MuseScore" tip:"Invoke MuseScore on score XML" args:[/home/aumhren/apps/musescore/MuseScore-Studio-4.4.4.243461245-x86_64.AppImage, {}]}
```
i tried various permutations of above plugin code, including running from shell ...

``` sh
$ ./MuseScore-Studio-4.4.4.243....AppImage
```
will open musescore, as expected

how should i code 'plugins.xml' so it will open musescore as default plugin ?

linux mint 21.3

**Comment by hbitteur** [comment]
Two things:

1. There is a typo in printing out the error. It's in line 369 of class org.audiveris.omr.plugin.Plugin.java

instead of: 
> logger.warn("Error launching {} {}" + this, ex.toString(), ex);

it should be;
> logger.warn("Error launching {} {}", this, ex.toString(), ex);

No big deal. I just pushed the commit 9c99689723b9ac79638d96f88867c1dc2ada9ce3 on the development branch, so you can simply pull and build.

2. Now, the error itself. The message tells something like:
> Cannot run program "/home/aumhren/apps/musescore/MuseScore-Studio-4.4.4.243461245-x86_64.AppImage"
> No such file or directory

Could you make sure that this file does exist, and can be launched via the command "/home/aumhren/apps/musescore/MuseScore-Studio-4.4.4.243461245-x86_64.AppImage"
Because launching it via the command "./MuseScore-Studio-4.4.4.243....AppImage" is different: you are launching it as a sibling program to be found in your current working directory.

In other words:
>$  /home/aumhren/apps/musescore/MuseScore-Studio...
>$ ./MuseScore-Studio...

Are different commands and might lead to different results.

Please use the command "pwd" to check where you are.

**Comment by aum7** [comment]
yes, first thing one would usually do is make sure file exists, and yes, i tried
``` sh
$ /home/myfolder/apps/musescore/MuseScore....AppImage
```
and it works as expected - *musescore runs*

i just did not wrote it, my bad, since for me it is obvious thing to do
i also use [tab] inside terminal, it will suggest only existing paths & files

i have git-fobia (messed with 'reset --hard' and was punished) and will probably wait for release

thx for quick reply

have fun

**Comment by aum7** [comment]
i am not sure
if i pull & build latest development
(assuming correct path & file exists) should it work - invoking 'default plugin' ?
``` sh
$ java -version
```
on my system returns
`openjdk version "11.0.25" 2024-10-15`
while docs tell a story of `17` & `21 jdk`

can i use this `openjdk` version to build ?

**Comment by hbitteur** [comment]
The default plugin is the one which appears first in your `plugins.xml` file. Since you have only one plugin element, it is the default one.

Regarding Java, you will need Java 21 for the development branch. And yes, I think openjdk 21 will do the job.

**Comment by hbitteur** [comment]
@aum7 
Is your plugin working correctly now?

**Comment by aum7** [comment]
no
i believe i cloned the latest dev, via terminal,
and tried compiling
following all errors, build by build,
i installed needed packages
but still failed to build
so i abandoned the build

will wait for release

version on my linux mint is installed via
org.audiveris.audiveris.flatpakref

anyway, currently i just double-click to open the .mxl file in musescore
and continue from there

thx for sharing

ps: i am a bit confused about java, since my installed version (`$ java -version`) is '11.0.25' (2024-10-15) - considering the date, i believe it is fairly recent version of java; though i just checked the net, and there is `openjdk 23` available; i do not want to just install / update, it might break my mint os; therefore best solution would be to use `vm` and install & build there, but am too lazy, and since **audiveris (ole!)** works just fine, i actually have no need

**Comment by hbitteur** [comment]
I don't know what "(2024-10-15) " is, it sounds like an installation date.

Anyway, Java 11 is **not** Java 21.
You should install openjdk21, otherwise you won't be able to build Audiveris
